docs/api/csf/csf-factories.mdx

```diff
@@ -0,0 +1,305 @@
+---
+title: 'Component Story Format (CSF)'
+isTab: true
+tab:
+  order: 2
+  title: CSF Factories (Experimental)
+---
+
+<If notRenderer="react">
+
+<Callout variant="info">
+  CSF Factories are currently only supported in [React](?renderer=react) projects.
+</Callout>
+
+</If>
+
+<If renderer="react">
+
+<Callout variant="warning" icon="🧪">
+  This is an experimental feature and (though unlikely) the API may change in future releases. We welcome feedback and contributions to help improve this feature.
+</Callout>
+
+CSF Factories are the next evolution of Storybook's Component Story Format (CSF). This new API uses a pattern called factory functions to provide full type safety to your Storybook stories, making it easier to configure addons correctly and unlocking the full potential of Storybook's features.
+
+This reference will provide an overview of the API and a migration guide to upgrade from CSF 3.
+
+## Overview
+
+The CSF Factories API is composed of four main functions to help you write stories. Note how the functions operate as a factory, each producing the next function in the chain (`definePreview` → `preview.meta` → `meta.story`), providing full type safety at each step.
+
+### `defineMain`
+
+With CSF Factories, your [main Storybook config](../main-config/main-config.mdx) is specified by the `defineMain` function. This function is type-safe and will automatically infer types for your project.
+
+{/* prettier-ignore-start */}
+<CodeSnippets path="csf-factories-main-example.md" />
+{/* prettier-ignore-end */}
+
+### `definePreview`
```

Comment:
would it make sense to mention in this section that now the preview object will be used to define stories, and therefore will be imported in every story file?